lib/rack/session/redis.rb

```diff
@@ -14,7 +14,9 @@ def initialize(app, options = {})
         super
 
         @mutex = Mutex.new
-        @pool = ::Redis::Store::Factory.create @default_options[:redis_server]
+        @pool = @default_options.has_key?(:redis_store) ?
```

Comment:
This line is rather long. I think it'd be better split out over multiple lines and you could use `fetch` as well:

```ruby
@pool = @default_options.fetch(:redis_store,
  ::Redis::Store::Factory.create(@default_options[:redis_server]))
```